File: problems/387.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        d = {}
        for c in s:
            if c not in d:
                d[c] = 1
            else:
                d[c] += 1
        for k,v in d.items():
            if v == 1:
                return s.index(k)
        return -1

File: problems/387.cpp
class Solution {
public:
    std::unordered_map<char, int> m;
    int firstUniqChar(string s) {
        for (auto &c : s) {
            m[c]++;
        }
        for (int i = 0; i < s.size(); i++) {
            if (m[s[i]] == 1) return i;
        }
        return -1;
    }   
};